rosetta-lang/src/main/java/com/regnosys/rosetta/validation/RosettaSimpleValidator.xtend

```diff
@@ -203,12 +204,12 @@ class RosettaSimpleValidator extends AbstractDeclarativeRosettaValidator {
 
 	@Check
  	def void switchArgumentTypeMatchesCaseStatementTypes(SwitchOperation op) {
- 		val argumentRType = op.argument.RType
+ 		val argumentRType = op.argument.RMetaAnnotatedType
```

Comment:
This one should only consider the RType - otherwise the next check `isSubtypeOf` will fail for input with metadata.